src/components/Api.js

```diff
@@ -0,0 +1,160 @@
+export default class Api {
+  constructor(data) {
+    this.baseUrl = data.baseUrl;
+    this.headers = data.headers;
+  }
+
+  // Method to get initial cards
+  getInitialCards() {
+    return fetch(`${this.baseUrl}/cards`, {
+      headers: this.headers,
+    })
+      .then((res) => {
+        if (res.ok) {
+          return res.json();
+        }
+        return Promise.reject(`Error: ${res.status}`);
+      })
+      .catch((err) => {
+        console.error("Failed to fetch initial cards:", err);
```

Comment:
You should **not** use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request, and this end is in `index.js`  after all `then` blocks. And in real projects the `catch` blocks usually open error popups and so on.